code/obj/machinery/glass_recycler.dm

```diff
@@ -175,12 +177,15 @@ TYPEINFO(/obj/machinery/glass_recycler)
 
 		if (src.glass_amt < target_product.product_cost)
 			src.visible_message("<span class='alert'>[src] doesn't have enough glass to make that!</span>")
+			playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 40, 0, 0.1)
 			return
 
 		var/obj/item/G = new target_product.product_path(get_turf(src))
 		src.glass_amt -= target_product.product_cost
 
 		src.visible_message("<span class='notice'>[src] manufactures \a [G]!</span>")
+		playsound(src.loc, 'sound/machines/vending_dispense_small.ogg', 40, 0, 0.1)
+		usr.put_in_hand_or_eject(G)
```

Comment:
Ideally add a new `mob/user` argument to this proc and pass the user explicitly, then use that. The `usr` var can backfire in various fun ways.

Also use the `?.` operator instead of `.` to handle the case when `usr`/`user` is null.